README.md

```diff
@@ -25,3 +25,56 @@ sudo chmod 666 /dev/ttyUSB0
 sudo chown $USER /dev/ttyUSB0
 sudo adduser $USER dialout
 ```
+# System Design
+
+## Modules
+Communication between the Hardware and Game Event Loop modules is done via message queue.
+Hardware events are placed in the queue by the Hardware module then the Game is ticked, this causes the Game to process the events, using the Player Engine to update the state of the game and enqueues Game messages into a queue for the Hardware system to process (resulting in animations to confirm actions were taken).
+The Game Event Loop uses functions in the player engine to update the state of the game.
+```mermaid
+graph
+    A[Hardware] --> B[(Game Message Queue)]
+    B --> C[Game Event Loop]
+    C --> D[(Hardware Message Queue)]
+    D --> A
+    C -..-> E[Player Engine]
+```
+
+### Persistent player state
+The persistent game state is stored in special hardware memory represented by a struct from the Player Engine.
+When the Hardware ticks the Game it sends the current player state.
+Then the Game calls functions from the Player Engine to transform the state based on the events it received from hardware.
+Once all messages are processed the persistent state is passed back to the hardware and the new state is stored in the special memory before sleep.
+
+### Hardware Module
+The hardware subsystem performs these steps on each wake-up:
+```mermaid
+graph TD
```

Comment:
Is it relevant here to note https://github.com/kestrel-labs/superspreader/blob/main/arduino/health_monitor/health_monitor.ino#L240 or do you think that's just an implementation detail?